design/ceph-device-set.md

```diff
@@ -0,0 +1,86 @@
+# Device Sets for rook-ceph
+
+**Target version**: Rook 1.1
+
+## Background
+The primary motivation for this feature is to take advantage of the mobility of storage in cloud-based environments. In environments like AWS you can request remote storage that can be dynamically provisioned and attached to any node in a given Availability Zone (AZ). If rook-ceph were able to take advantage of this it would improve the resiliency of the cluster and further automate disaster recovery. In addition, the design can also accommodate non-cloud environments through the use of local PVs.
+
+## CephDeviceSet CRD
+```yaml
+apiVersion: "ceph.rook.io/v1"
+kind: CephDeviceSet
+metadata:
+  name: my-device-set
+  namespace: rook-ceph
+spec:
+  # The Ceph cluster these devices belong to
+  cluster: my-ceph-cluster
```

Comment:
specifying the cluster wouldn't be necessary since the namespace uniquely identifies a cluster